source/tutorials/file-sets.md

```diff
@@ -0,0 +1,661 @@
+  (and equivalently [`lib.sources.cleanSourceWith`](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.sources.cleanSourceWith))
+  can address these problems.
+  However, it's often hard to express the desired path selection using the `filter` function interface.
+
+In this tutorial you'll learn how to use Nixpkgs' [`lib.fileset` library](https://nixos.org/manual/nixpkgs/unstable/#sec-fileset) to work with local files in derivations.
+It abstracts over built-in functionality with a safer and more convenient interface.
+
+## File sets
+
+The basic concept is that of a _file set_,
+a data type representing a collection of local files.
+File sets can be created, composed, and manipulated with the various functions of the library.
+
+You can explore and learn about the library with [`nix repl`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-repl):
+
+```shell-session
+$ nix repl -f https://github.com/NixOS/nixpkgs/tarball/master
+...
+nix-repl> fs = lib.fileset
+```
+
+:::{note}
+We're using the `master` branch here, because the covered library features are part of NixOS 23.11 and up, which has not been released at the time of writing.
+:::
+
+The [`trace`](https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.fileset.trace) function pretty-prints the files included in a given file set:
+
+```shell-session
+nix-repl> fs.trace ./. null
+trace: /home/user (all files in directory)
+null
+```
+
+All functions that expect a file set for an argument also accept a [path](https://nixos.org/manual/nix/stable/language/values#type-path) instead.
+Such path arguments are then [implicitly turned into sets](https://nixos.org/manual/nixpkgs/unstable/#sec-fileset-path-coercion) that contain _all_ files under the given path.
+In the previous trace this is indicated by `(all files in directory)`.
+
+:::{tip}
+The `trace` function pretty-prints its first argument and returns its second argument.
+But since you often just need the pretty-printing in `nix repl`, you can omit the second argument:
+
+```shell-session
+nix-repl> fs.trace ./.
+trace: /home/user (all files in directory)
+«lambda @ /nix/store/1czr278x24s3bl6qdnifpvm5z03wfi2p-nixpkgs-src/lib/fileset/default.nix:555:8»
+```
+:::
+
+Even though file sets conceptually contain local files, these files are *never* added to the Nix store unless explicitly requested.
+So You don't have to worry as much about accidentally copying secrets into the world-readable store.
+
+In this example, although we pretty-printed the home directory, no files were copied.
+This is in contrast to coercion of paths to strings such as in `"${./.}"`,
+which copies the whole directory to the Nix store on evaluation!
+
+:::{warning}
+When using [experimental Flakes](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake), local directories with a `flake.nix` are always copied into the Nix store *completely* unless it is a Git repository!
```

Comment:
What are "experimental Flakes"? Are there non-experimental flakes?

I still think that should be "With the `flakes` experimental feature enabled, ..."

Reply:
> When using the [`flakes` and `nix-command` experimental features](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake),
a local directory within a Flake is always copied into the Nix store *completely* unless it is a Git repository!

In the meeting